lib/pool.js

```diff
@@ -365,10 +365,22 @@ SMTPConnectionPool.prototype._onConnectionEnd = function(connection) {
         }
     }
 
+    // Call the current message's callback if one still exists
+    // This is important to do here and in _onConnectionError because
+    // _onConnectionEnd() can be called without _onConnectionError being called
```

Comment:
What happens without this code?

Reply:
Updated the comment to include that info. tl;dr is that callbacks to sendMail could be left hanging.